worlds/dark_souls_3/__init__.py

```diff
@@ -1442,81 +1442,81 @@ def post_fill(self):
         items, later spheres get higher-level ones. Within a sphere, items in DS3 are distributed in
         region order, and then the best items in a sphere go into the multiworld.
         """
-
-        locations_by_sphere = [
-            sorted(loc for loc in sphere if loc.item.player == self.player and not loc.locked)
-            for sphere in self.multiworld.get_spheres()
-        ]
-
-        # All items in the base game in approximately the order they appear
-        all_item_order: List[DS3ItemData] = [
-            item_dictionary[location.default_item_name]
-            for region in region_order
-            # Shuffle locations within each region.
-            for location in self._shuffle(location_tables[region])
-            if self._location_status(location).is_randomized
-        ]
-
-        # All DarkSouls3Items for this world that have been assigned anywhere, grouped by name
-        full_items_by_name: Dict[str, List[DarkSouls3Item]] = defaultdict(list)
-        for location in self.multiworld.get_filled_locations():
-            if location.item.player == self.player and (
-                location.player != self.player or self._location_status(location).is_randomized
-            ):
-                full_items_by_name[location.item.name].append(location.item)
-
-        def smooth_items(item_order: List[Union[DS3ItemData, DarkSouls3Item]]) -> None:
-            """Rearrange all items in item_order to match that order.
-
-            Note: this requires that item_order exactly matches the number of placed items from this
-            world matching the given names.
-            """
-
-            # Convert items to full DarkSouls3Items.
-            converted_item_order: List[DarkSouls3Item] = [
-                item for item in (
-                    (
-                        # full_items_by_name won't contain DLC items if the DLC is disabled.
-                        (full_items_by_name[item.name] or [None]).pop(0)
-                        if isinstance(item, DS3ItemData) else item
-                    )
-                    for item in item_order
-                )
-                # Never re-order event items, because they weren't randomized in the first place.
-                if item and item.code is not None
+        if self.options.smooth_upgrade_items or self.options.smooth_soul_items or self.options.smooth_upgraded_weapons:
```

Comment:
Can't you just early return if this isn't true to avoid a bunch of extra indentation?

Reply:
Gosh you are SO right. I'll fix that tomorrow

Reply:
Fixed now